src/main/java/dog/pawbook/model/managedentity/owner/UniqueEntityList.java

```diff
@@ -85,13 +85,15 @@ private int genID() {
      * Adds an entity to the list.
      * The entity must not already exist in the list.
      */
-    public void add(Entity toAdd) {
+    public int add(Entity toAdd) {
         requireNonNull(toAdd);
 
         if (contains(toAdd)) {
             throw new DuplicateEntityException();
         }
-        internalList.add(new Pair<>(genID(), toAdd));
+        int idEntity = genID();
+        internalList.add(new Pair<>(idEntity, toAdd));
+        return idEntity;
```

Comment:
Naming is especially confusing for this part, since idEntity supposedly refers to a pair already.

Reply:
I agree, will change it!